test/local/password_check_test.js

```diff
@@ -17,6 +17,9 @@ var MockCustoms = {
 
 var MockDB = {
   locked: {},
+  checkPassword: function (uid) {
+    return uid === 'correct_password';
+  },
```

Comment:
It is called by checkPassword ('../../lib/routes/utils/password_check') which now calls the database for password comparison checking (it was the point of this patch after all haha)